### DISCUSSION TITLE
PATCH w product-offers nie działa, gdy stock: available 0

### DISCUSSION THREAD

**Discussion by buehehe666** [allegro/allegro-api#5565] [Suggestions - resolved]
Witam, napotkałem na problem, którego nie potrafię rozwiązać przy pomocy forum oraz dokumetacji. Proszę o podpowiedź co robię nie tak.

Zrobiłem skrypt na aktualizację stanów magazynowych. Korzystam z metody PATCH w /sale/product-offers/.

Wszystko działa super oprócz przypadku, gdy próbuję zmienić dostępny zapas na 0 (stock: { available: 0 }). Każda inna liczba w kombinacji z pozostałymi parametrami (tj. external - id, sellingMode - price - amount) działa. 
Gdy ustawiam stock - available  = 0 i np. zmieniam status na ENDED, to status zmienia się poprawnie, stock pozostaje bez zmian, a odpowiedź serwera dostaję undefined. Gdy stock jest inną liczbą większą od zera, wszystko działa i otrzymuję odpowiedź o prawidłowej aktualizacji oferty.

Według dokumentacji pole stock - available powinno być: integer <int32> >= 0, a z jakiegoś powodu nie chce mi to działać.

Czy powinienem wysłać w zapytaniu jakiś określony zestaw parametrów, żeby można było zmienić stock - available na 0?

**Comment by hahyes** [comment]
Widocznie jest błąd w dokumentacji albo dotyczy to wyświetlanych danych a nie podawanych w zapytaniu. Nie wyobrażam sobie sytuacji, gdzie aukcja ma zapas = 0. To nie jest magazyn.

**Reply by buehehe666** [reply]
![image](https://user-images.githubusercontent.com/102182351/159669361-9c126c8c-7c16-430d-b0c0-b6ac393cc57f.png)
![image](https://user-images.githubusercontent.com/102182351/159669851-97c2ab44-f5e9-44ce-97d3-431ceda4c454.png)
****
Teoretycznie w widoku moich ofert też nie pokazuje 0 przy zakończonych ofertach. Nie jestem w tym ekspertem, bo powiedzmy, że jestem na początku swojej drogi w tych tematach, ale spodziewałem funkcji, która np. zamknie ofertę, gdy ilość spadnie poniżej 1, wzorem automatycznego zamykania ofert z wykupionymi wszystkimi sztukami.

W każdym razie dobrze byłoby wiedzieć, czy mam jeszcze coś z tym kombinować, czy użyć innego sposobu na osięgnięcie celu.

**Reply by Lukasz-Zurek** [reply]
Konsultuję to z zespołem odpowiedzialnym za tę funkcjonalność. Proszę o cierpliwość.

**Reply by hahyes** [reply]
> Teoretycznie w widoku moich ofert też nie pokazuje 0 przy zakończonych ofertach. Nie jestem w tym ekspertem, bo powiedzmy, że jestem na początku swojej drogi w tych tematach, ale spodziewałem funkcji, która np. zamknie ofertę, gdy ilość spadnie poniżej 1, wzorem automatycznego zamykania ofert z wykupionymi wszystkimi sztukami.

Możliwe, że był taki zamysł przy projektowaniu zasobu, ale dalsza walidacja danych wyklucza taką możliwość. Musisz po prostu zamykać ofertę zamiast aktualizować 'available' na 0, to na pewno zadziała. Osobiście nigdy nie zmieniałem `stock` w taki sposób jak ty, więc mogę tylko przypuszczać.

**Comment by MartaNowaczyk** [comment]
[Wdrożyliśmy](https://github.com/allegro/allegro-api/issues/7280) możliwość zakończenia oferty poprzez ustawienie "stock.available" na "0".
